### PR TITLE
End to end eval

### PR DESCRIPTION
**Pull request by hooshmandshr**
Adding the following files:
- yass/evaluate/analyzer.py
- yass/evaluate/visualization.py
- yass/evaluate/util.py

Allows for evaluating accuracy stability be done using just config file.

Saves intermediate results, no need for rerunning these metrics every time.

Visualizations for metric vs. snr, curve plots and spatial traces of templates gold standard vs. yass.

### CODE REVIEW

File: src/yass/evaluate/analyzer.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+from yass.pipeline import run
+
+
+class Analyzer(object):
+    """Class that analyzes the performance of yass on a certain dataset"""
+
+    def __init__(self, config, gold_std_spike_train=None):
+        """Sets up the analyzer object with configurations.
+
+        Parameters:
```
Comment (edublancas) [inline src/yass/evaluate/analyzer.py]: can you update this to be numpydoc-style?

File: src/yass/evaluate/stability.py [hunk 2]
```diff
@@ -62,7 +62,10 @@ def __init__(self, rec_file, geom_file, sample_rate,
         self.n_batches = n_batches
         self.n_chan = n_chan
         self.radius = radius
-        self.geometry = np.genfromtxt(geom_file, delimiter=' ')
+        if geom_file.endswith('.txt'):
```
Comment (edublancas) [inline src/yass/evaluate/stability.py]: there is one function in the geometry module for this, it handles more geometry types

**Comment by hooshmandshr** [comment]
@edublancas: How to link to examples if they are python notebooks?

**Comment by edublancas** [comment]
are the notebooks in the yass examples repo?

**Comment by hooshmandshr** [comment]
No, but I can add it. It makes reference to some paths inside our machine. That's why I haven't added it yet.

**Comment by edublancas** [comment]
Ok, might be a good idea to just have placeholders and then if anyone wants to use them just replace by the real path

**Pull request merged**